client/src/main/scala/skuber/api/Configuration.scala

```diff
@@ -143,23 +143,59 @@ object Configuration {
             certificateAuthority=pathOrDataValueAt(clusterConfig, "certificate-authority","certificate-authority-data"))
 
 
-        val k8sClusterMap = topLevelYamlToK8SConfigMap("cluster", toK8SCluster _)
-
-        def toK8SAuthInfo(userConfig:YamlMap):AuthInfo = AuthInfo(
-          clientCertificate = pathOrDataValueAt(userConfig, "client-certificate", "client-certificate-data"),
-          clientKey = pathOrDataValueAt(userConfig, "client-key", "client-key-data"),
-          jwt = userConfig.containsKey("auth-provider") match {
-            case true => val authProvider = child(userConfig,"auth-provider"); authProvider.containsKey("config") match {
-              case true => val config = child(authProvider,"config"); optionalValueAt(config,"id-token")
-              case false => None
+        val k8sClusterMap = topLevelYamlToK8SConfigMap("cluster", toK8SCluster)
+
+        def toK8SAuthInfo(userConfig:YamlMap): AuthInfo = {
+
+          def authProviderRead(authProvider: YamlMap): Option[AuthProviderAuth] = {
+            val config = child(authProvider, "config")
+            name(authProvider).toLowerCase match {
+              case "oidc" =>
+                Some(OidcAuth(idToken = valueAt(config, "id-token")))
+              case "gcp" =>
+                Some(GcpAuth(accessToken = valueAt(config, "access-token")))
+              case _ => None
             }
-            case false => None
-          },
-          token = optionalValueAt(userConfig, "token"),
-          userName = optionalValueAt(userConfig, "username"),
-          password = optionalValueAt(userConfig, "password"))
+          }
+
+          val maybeAtuh = optionalValueAt[YamlMap](userConfig, "auth-provider") match {
```

Comment:
I assume  'maybeAtuh' is a typo, should be 'maybeAuth'?

Reply:
Yeah, thanks. Damn phat phingers